### ISSUE TITLE
Bounds on SDP matrix variables 

### ISSUE THREAD

**Issue by harshangrjn** [JuliaOpt/JuMP.jl#1945]
In JuMP v0.18.5: 

Given an SDP matrix `@variable(m, X[1:3,1:3], SDP)`, when I enforce bounds on the matrix elements, say 1 <= X[1,2] <= 2 using setlowerbound and setupperbound, I get this error as the problem seems to loose the SDP structure.  
```
ERROR: AssertionError: lb == -Inf && ub == Inf
```
But enforcing the above mentioned bounds via constraints seem to be fine. Is this expected for SDPs?

**Comment by blegat** [comment]
Which solver are you usiing ?

**Comment by harshangrjn** [comment]
@blegat Mosek. But I think this is happening even if I do not initialize any solver, i.e, m = Model()

**Comment by odow** [comment]
Does it work on 0.19?

**Comment by blegat** [comment]
Mosek does not support bounds on SDP variables. The error message could be improved though. If there is no solver, then I would be surprised that you see an error

**Comment by ccoffrin** [comment]
@harshangrjn what you suggest is working in PowerModels (e.g. see https://github.com/lanl-ansi/PowerModels.jl/blob/master/src/form/wrm.jl), I have tested with Mosek on JuMP v0.18.  I have not checked if there is a regression on JuMP v0.19 with Mosek, but it works with SCS for sure.

**Comment by blegat** [comment]
With SCS, all variables are free so the `SingleVariable` constraints are transformed to `ScalarAffineFunction` constraints.
With Mosek it shouldn't work neither on JuMP v0.18 and JuMP v0.19.
I would recommend using `@constraint(model, x[i, j] <= ub)` instead of `@constraint(model, x[i, j] in MOI.LessThan(ub))` so that it is a `ScalarAffineFunction` instead of a `SingleVariable` one.

**Comment by harshangrjn** [comment]
@blegat Thanks for the clarifications. Error doesn't appear without initializing a solver. I may have run the problem in the terminal without overloading the model. I can close this issue if this error doesn't anymore exist in JuMP 0.19. 

@ccoffrin  Either with Mosek or SCS and JuMP 0.18.5, the assertion error persists while using setlowerbound and setupperbound. Here is a simple problem: 
```
#m = Model(solver=MosekSolver())
m = Model(solver=SCSSolver())
@variable(m, X[1:3,1:3], SDP)
setlowerbound(X[1,2], -0.2)
setupperbound(X[1,2], -0.1)
setlowerbound(X[2,3], 0.4)
setupperbound(X[2,3], 0.5)
@objective(m, Max, X[1,3])
@constraint(m, [i=1:3], X[i,i] == 1)
```

**Comment by frederikgeth** [comment]
Just pointing out that PowerModels never defines SDP variables, only @SDconstraints on matrices composed of (normal) variables. Have you tried that approach?

**Comment by mlubin** [comment]
I don't see anything that needs to be fixed on JuMP 0.19. Please reopen if this issue isn't resolved.
